taf/git.py

```diff
@@ -26,22 +31,24 @@ def __init__(
       additional_info: a dictionary containing other data (optional)
       default_branch: repository's default branch
     """
-        self.repo_path = str(Path(repo_path).resolve())
+        self.path = str(Path(repo_path).resolve())
```

Comment:
What do you think about changing this to `self._path = Path(repo_path).resolve()` and adding a property called `path`? There are some places in code where we could use `_path` as _pathlib.Path_ instance (comments below). That way we will keep backwards compatibility.

Reply:
I think that that makes sense. I think that one the reason why we couldn't use `pathlib`'s path was because of `Python 3.5` issues. Since we decided not to support `Python 3.5`, we can transition to using `pathlib`'s path where that is possible. We could keep that in mind while working on other issues. I add this now though